libs/core/tests/unit_tests/test_tools.py

```diff
@@ -2468,3 +2468,33 @@ def foo(x: int) -> str:
     assert foo.invoke({"type": "tool_call", "args": {"x": 0}, "id": ""}) == ToolMessage(
         content="hi", name="foo", tool_call_id=""
     )
+
+
+def test_tool_decorator_description() -> None:
+    @tool
+    def foo(x: int) -> str:
+        """Foo."""
+        return "hi"
+
+    @tool(description="description")
+    def foo_description(x: int) -> str:
+        """Foo."""
+        return "hi"
+
+    class ArgsSchema(BaseModel):
+        """Bar."""
+
+        x: int
+
+    @tool(args_schema=ArgsSchema)
+    def foo_args_schema(x: int) -> str:
+        return "hi"
+
+    @tool(description="description", args_schema=ArgsSchema)
+    def foo_args_schema_description(x: int) -> str:
+        return "hi"
+
+    assert foo.description == "Foo."
+    assert foo_description.description == "description"
+    assert foo_args_schema.description == "Bar."
+    assert foo_args_schema_description.description == "description"
```

Comment:
could we validate the descriptino of the tool that's produced when we generate it's json schema?